### DISCUSSION TITLE
Single Custodian Policy Among Multiple Accounts

### DISCUSSION THREAD

**Discussion by pavantheavenger** [cloud-custodian/cloud-custodian#7563] [AWS] [answered]
Hello Team,

Can you please help me understand how to use a Single Cloud Custodian Policy in an Organization.

I want to use Lambda (Python) Function for the same, please suggest some URL or Code for reference.

Where I can use the same lambda Function among all the accounts so that it would be easy for managing the Function.

Expecting to here from you soon.

Thank You !

Regards

Pavan Kumar

**Comment by ajkerrigan** [comment, marked as answer]
Hi @pavantheavenger - Custodian does support this sort of pattern but it requires some custom setup on your end. You may find the discussion in https://github.com/cloud-custodian/cloud-custodian/discussions/7448 helpful since it has the same goal.

**Reply by pavantheavenger** [reply]
Hi @ajkerrigan 

Thank you very much for the response.

Can you please correct my python script below:
---------------------------------------------------------------------
import os
import logging
import sys
import boto3
import c7n_org
from c7n_org import cli as org

logger = logging.getLogger()
logger.setLevel(logging.INFO)

output_directory = 'Output'

assumed_role = 'arn_of_IAM_Role'
config_filename = 'accounts.yml'
ec2_filename = 'policies1.yml'
sg_filename = 'policies2.yml'

default_c7n_config = {
    'skip-validation': True,
    'vars': 'policies/Python/var.yml',
    'debug': True,
    'assume': assumed_role,
    'output_dir': os.path.join(output_directory, 'output'),
    'region': 'policies/Python/region.yml',
    'config': [config_filename],
    'policies': [ec2_filename, sg_filename]
}

run_config = c7n_org.cli.Config.empty(**default_c7n_config)
print(run_config)

logger.info('Running policy: ' + ec2_filename + sg_filename)

print(logger.info)

c7n_org.cli.run(run_config)

-------------------------------------------------------------------------------

Error Msgs :
Error: Missing option '-c' / '--config'.

Please help me out with this, expecting to hear from you soon.

Thank You !

Regards 

Pavan

**Reply by pavantheavenger** [reply]
Else please help me at least the module/function name which I can use to disclose the values of the config, policy, output accordingly.

Expecting to hear from you soon.

Thank You !

Regards 

Pavan

**Reply by ajkerrigan** [reply]
You'd typically [generate a config file](https://cloudcustodian.io/docs/tools/c7n-org.html#config-file-generation) using something like [orgaccounts.py](https://github.com/cloud-custodian/cloud-custodian/blob/master/tools/c7n_org/scripts/orgaccounts.py), and then pass the result in as `-c` or `--config` to your `c7n-org run` command as described in [Running a Policy with c7n-org](https://cloudcustodian.io/docs/tools/c7n-org.html#running-a-policy-with-c7n-org). Scripting against the underlying Python module is going a bit off the supported path, because:
* Custodian doesn't provide a stable public interface aside from the CLI
* Any documentation or examples assume that you'll be running `c7n-org run...`

It's possible to get it working from the Python side, but it'd be a bit messy and likely to break without warning in the future.

**Comment by pavantheavenger** [comment]
Hello AJ Kerrigan,

Thank you for the quick response.

Can you please suggest me the what is function that I need to use in c7n_org to consume the values of config, policies etc.

Please help me out with the name of that function which can consume the value of --config in module c7n_org.

Expecting to hear from you soon. 

Thank You !

Regards

Pavan
